DataFormats/simulation/src/Stack.cxx

```diff
@@ -270,7 +278,7 @@ void Stack::PushTrack(int toBeDone, TParticle& p)
     mPrimaryParticles.push_back(p);
```

Comment:
@sawenzel 
I think the issue with G4 originates from here.
The particle is added to the vector before changing the kToBeDone status.

Reply:
will modify and check if this fixes it